smhi/metobs.py

```diff
@@ -36,76 +41,109 @@ def __init__(self, type: str = "json"):
+            version = "1.0"
+
         if version != "1.0":
-            raise Exception("Only supports version = 1.0.")
+            raise NotImplementedError(
+                "Version {} not supported. Only supports version = 1.0.".format(version)
+            )
 
         self.version = version
-        self.parameter = SMHIParameterV1(self.type, self.available_versions, version)
+        self.parameter = MetObsParameterV1(self.available_versions)
 
     def fetch_stations(self, parameter: str = None, parameter_title: str = None):
         """
-        Fetch SMHI API (version 1) stations from given parameter.
+        Fetch SMHI MetObs API (version 1) stations from given parameter.
 
         Args:
             parameter: id of data
             parameter_title: exact title of data
         """
         if parameter is None and parameter_title is None:
-            raise Exception("Both arguments None.")
-
-        if self.parameter is None:
-            raise Exception("Fetch parameters first.")
-
-        if parameter_title is None:
-            self.station = SMHIStationV1(
-                self.type, self.parameter.resource, parameter=parameter
-            )
+            raise NotImplementedError("Both arguments None.")
 
-        if parameter is None:
-            self.station = SMHIStationV1(
-                self.type, self.parameter.resource, parameter_title=parameter_title
+        if parameter:
+            self.station = MetObsStationV1(self.parameter.resource, parameter=parameter)
+        else:
+            self.station = MetObsStationV1(
+                self.parameter.resource, parameter_title=parameter_title
             )
 
-    def fetch_periods(self, station: str = None, station_set: str = None):
+    def fetch_periods(self, station: str = None, stationset: str = None):
         """
-        Fetch SMHI API (version 1) periods from given station.
+        Fetch SMHI MetObs API (version 1) periods from given station.
 
         Args:
             station: id of data
-            station_set: exact title of data
+            stationset: exact title of data
         """
-        if station is None and station_set is None:
-            raise Exception("Both arguments None.")
-
-        if self.station is None:
-            raise Exception("Fetch stations first.")
+        if station is None and stationset is None:
+            raise NotImplementedError("Both arguments None.")
 
-        if station_set is None:
-            self.period = SMHIPeriodV1(self.type, self.station.station, station=station)
-
-        if station is None:
-            self.period = SMHIPeriodV1(
-                self.type, self.station.station, station_set=station_set
-            )
+        if station:
+            self.period = MetObsPeriodV1(self.station.station, station)
+        else:
+            self.period = MetObsPeriodV1(self.station.station, stationset=stationset)
 
     def fetch_data(self, period: str = "corrected-archive"):
```

Comment:
def get_data(self, period: str = "corrected-archive"):

Reply:
There is another "get_data" below, hence the double naming must be resolved before removing the "fetch" terminology

Reply:
Enough.